libvast/src/system/node.cpp

```diff
@@ -362,21 +363,45 @@ node_state::spawn_command(const invocation& inv,
     spawn_inv.options["import"] = import_opt;
     caf::put(spawn_inv.options, "vast.import", import_opt);
   }
-  // Spawn our new VAST component.
-  spawn_arguments args{spawn_inv, this_node->state.dir, label};
-  auto component = spawn_component(spawn_inv, args);
-  if (!component) {
-    if (component.error())
-      VAST_WARNING(__func__,
-                   "failed to spawn component:", render(component.error()));
-    rp.deliver(component.error());
-    return caf::make_message(component.error());
+  auto doit = [=](taxonomies_ptr t) mutable {
+    if (t) {
+      auto expr = normalized_and_validated(spawn_inv.arguments);
+      if (!expr)
+        rp.deliver(expr.error());
+      auto resolved = resolve(*t, *expr);
+      spawn_inv.arguments = std::vector{to_string(resolved)};
+    }
+    // Spawn our new VAST component.
+    spawn_arguments args{spawn_inv, this_node->state.dir, label};
+    auto component = spawn_component(spawn_inv, args);
+    if (!component) {
+      if (component.error())
+        VAST_WARNING(
+          __func__, "failed to spawn component:", render(component.error()));
+      rp.deliver(component.error());
+      return;
+    }
+    this_node->monitor(*component);
+    auto okay = this_node->state.registry.add(*component, std::move(comp_type),
+                                              std::move(label));
+    VAST_ASSERT(okay);
+    rp.deliver(*component);
+    return;
+  };
+  if (std::set<std::string>{"counter", "eraser", "exporter", "pivoter"}.count(
+        comp_type)
+      > 0u) {
+    if (auto tr_ = this_node->state.registry.find_by_label("type_registry")) {
```

Comment:
We reserve `_` suffixes for member variables. I don't think this is need here; `tr` is not a keyword.